app/src/main/java/net/squanchy/favorites/view/NoFavoritesView.java

```diff
@@ -0,0 +1,55 @@
+package net.squanchy.favorites.view;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.LinearLayout;
+
+import net.squanchy.R;
+
+public class NoFavoritesView extends LinearLayout {
+
+    private static final int MAGIC_NUMBER_TO_TRIGGER_ACHIEVEMENT = 5;
+    private FloatingActionButton favoriteButton;
+
+    public NoFavoritesView(Context context) {
+        super(context);
+    }
+
+    public NoFavoritesView(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public NoFavoritesView(Context context, @Nullable AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        super.setOrientation(VERTICAL);
```

Comment:
You might want to override `setOrientation()` then and throw if the passed one is `HORIZONTAL`